frontend/src/routes/milestonegroup/+page.svelte

```diff
@@ -23,77 +23,232 @@
+			answer: null
+		}
+	];
+
 	// this is a list of data that can be fetched by the component later.
-	const milestonedata: any[] = [
+	const surveyData: any[] = [
 		{
 			header: 'Grobmotorik',
 			summary: 'something something',
 			image: imgJump,
 			href: `${base}/milestoneoverview`,
-			progress: 0.75
+			progress: 0.75,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Feinmotorik',
 			summary:
 				'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt',
 			image: null,
 			href: `${base}/milestoneoverview`,
-			progress: 1.0
+			progress: 1.0,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Geistige Grundfunktionen',
 			summary: 'something something',
 			image: imgHead,
 			href: `${base}/milestoneoverview`,
-			progress: 1.0
+			progress: 1.0,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Höhere Denkfunktionen',
 			summary: 'something something',
 			image: null,
 			href: `${base}/milestoneoverview`,
-			progress: 1.0
+			progress: 1.0,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Sprache',
 			summary: 'how much noise the child makes',
 			image: null,
 			href: `${base}/milestoneoverview`,
-			progress: 0.5
+			progress: 0.5,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Soziale Entwicklung',
 			summary:
 				'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit.',
 			image: imgHead,
 			href: `${base}/milestoneoverview`,
-			progress: 0.3
+			progress: 0.3,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Selbstregulation',
 			summary: 'something something',
 			image: null,
 			href: `${base}/milestoneoverview`,
-			progress: 0.6
+			progress: 0.6,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Emotionen',
 			summary: 'something something',
 			image: null,
 			href: `${base}/milestoneoverview`,
-			progress: 0.9
+			progress: 0.9,
+			milestoneData: milestoneData
 		},
 		{
 			header: 'Vorläuferfertigkeiten Schule',
 			summary:
 				'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit.',
 			image: null,
 			href: `${base}/milestoneoverview`,
-			progress: 0.3
+			progress: 0.3,
+			milestoneData: milestoneData
+		}
+	];
+
+	interface MilestoneData {
+		number: 4;
+		title: string;
+		desc: string;
+		observation: string;
+		help: string;
+		answer: string;
+	}
+
+	interface DataElement {
+		header: string;
+		summary: string;
+		milestoneData: any;
+		progress: number;
+		[key: string]: any;
+	}
+
+	function searchByStatus(data: DataElement[], dummy: any, key: string): DataElement[] {
```

Comment:
what is `dummy` for?

Reply:
needed because the underlying logic in `GalleryDisplay` allows for restriciting search to configurable columns/elements of the displayed gallery and thus needs an additional argument. This is not needed here and therefore uses a dummy value.

Reply:
ok... I refactored a bit and got rid of it.